ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java

```diff
@@ -299,35 +304,119 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
     };
   }
 
+  private void startTransaction(StreamInfo info, DataStreamRequestByteBuf request, ChannelHandlerContext ctx) {
+    try {
+      server.submitClientRequestAsync(info.getRequest()).thenAcceptAsync(reply -> {
+        if (reply.isSuccess()) {
+          ByteBuffer buffer = ClientProtoUtils.toRaftClientReplyProto(reply).toByteString().asReadOnlyByteBuffer();
+          sendReplySuccess(request, buffer, -1, ctx);
+        } else {
+          if (request.getType() == Type.STREAM_CLOSE) {
```

Comment:
it might be better to just use 
```
if (reply.isSuccess()) 
else if (request.getType() == Type.STREAM_CLOSE)
else if (request.getType() == Type.START_TRANSACTION)
else
```

Reply:
Updated.